app/repository/account.py

```diff
@@ -0,0 +1,26 @@
+from sqlalchemy.orm import Session
+from decimal import Decimal
+from typing import Any
+
+from model.account import Account
+
+
+def get_acc_by_id(db: Session, id: str) -> Any:
+    return db.query(Account).filter_by(account_id=id).first()
+
+
+def save_account(db: Session, id: str) -> Account:
+    acc = Account(account_id=id)
+
+    db.add(acc)
+    db.commit()
+
+    db.refresh(acc)
+    return acc
+
+
+def update_balance(db: Session, id: str, amount: Decimal) -> None:
+    acc = db.query(Account).filter_by(account_id=id).with_for_update().first()
```

Comment:
As our database picked is SQLite and we do not have features such as [Serializable Isolation level](https://www.postgresql.org/docs/9.1/transaction-iso.html#XACT-SERIALIZABLE) I've used a explicit locking feature, which in this case is `SELECT for UPDATE`.